### PR TITLE
adds k8s example

### PR DESCRIPTION
**Pull request by Nesurion**
Adds unit files to deploy k8s to coreos with inago. The k8s is a demo setup. It uses unsecure communication between components.
NOT a good idea to use this in production.

### CODE REVIEW

**Comment by Nesurion** [comment]
@giantswarm/cena LGTY and WFY?

**Comment by zeisss** [comment]
I guess we want a README showing the order of the steps to set this up.

File: example/k8s/k8s-apiserver.service [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[Unit]
+Description=Kubernetes API Server
+Documentation=https://github.com/GoogleCloudPlatform/kubernetes
```
Comment (zeisss) [inline example/k8s/k8s-apiserver.service]: :cake:

File: example/k8s/k8s-apiserver.service [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Requires=etcd2.service
+After=etcd2.service
+
+[Service]
+#EnvironmentFile=/etc/network-environment
```
Comment (zeisss) [inline example/k8s/k8s-apiserver.service]: delete?

**Comment by JosephSalisbury** [comment]
Yeah, WFM - i can start a replication controller, and scale it, it launches containers and everything.

I would like to see a little markdown document on what to do (submit, start, get kubectl, kubectl create -f something), it would make a good intro document

**Comment by puja108** [comment]
+1 on the .md maybe put it directly in the `docs` folder

**Comment by Nesurion** [comment]
@giantswarm/cena @puja108 RFR

File: docs/k8s.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# kubernetes
+
+With Inago you can easily deploy kubernetes on a coreos cluster.
```
Comment (JosephSalisbury) [inline docs/k8s.md]: on the whole, I'd refer to kubernetes as `Kubernetes`. The same with coreos, we should use `CoreOS`.

File: docs/k8s.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+The `example` folder contains a `k8s` example. Kubernetes is split into
+two groups: `k8s-master` and `k8s-node`.
+
+## note
```
Comment (JosephSalisbury) [inline docs/k8s.md]: I'd prefer to capitalise the section titles

File: docs/k8s.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+The deployment sets up a kubernetes cluster, which uses insecure communication
+between the components. Use this only as a Demo or POC. If you want to deploy a
+kubernetes cluster for production please modify the example, so a secure
```
Comment (JosephSalisbury) [inline docs/k8s.md]: use "so secure"

File: docs/k8s.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+communication is used. [Cluster TLS using OpenSSL](https://coreos.com/kubernetes/docs/latest/openssl.html)
+
+## deployment
+If your coreos cluster is up and running you can deploy kubernetes in 4 steps:
```
Comment (JosephSalisbury) [inline docs/k8s.md]: This needs to be run on a coreos box, right? it might be an idea to state that, just to be explicit

File: docs/k8s.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+1. submit the k8s-master by running: `inagoctl submit k8s-master`
+2. submit the k8s-node by running: `inagoctl submit k8s-node`
+3. start the k8s-master by running: `inagoctl start k8s-master`
+4. start the k8s-node by running: `inagoctl start k8s-node`
+
+And you're done!
+
+The k8s-node contains the `Global=true` flag and will be scheduled on all machines.
```
Comment (JosephSalisbury) [inline docs/k8s.md]: The k8s-node group's unit files contain the

File: docs/k8s.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+First we need the IP of the machine running the api-server. Login to a coreos machine
+and run
+
+```
+$ fleetctl list-units -fields=unit,machine --full --no-legend 2>/dev/null | grep ^k8s-master-apiserver.service | cut -d/ -f2 | paste -d, -s
+172.17.8.102
+```
+
+Now we can start a nginx example by running:
```
Comment (JosephSalisbury) [inline docs/k8s.md]: an nginx

File: docs/k8s.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+`kubectl -s 172.17.8.102:9090 run my-nginx --image=nginx --replicas=2 --port=80`
+
+To validate that the pods got indeed started, we can take a look at the pods.
```
Comment (JosephSalisbury) [inline docs/k8s.md]: To validate that the pods indeed got started,

File: docs/k8s.md (hunk 6 above)
Comment (Nesurion) [inline docs/k8s.md]: hm, the first paragraph states that this will deploy k8s on coreos. Feels to me like that is clear enough.

**Comment by JosephSalisbury** [comment]
LGTM

**Pull request merged**